cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/tools-control.tsx

```diff
@@ -1037,7 +1037,10 @@ export class ToolsControlComponent extends React.PureComponent<Props, State> {
                                     frame,
                                     occluded: false,
                                     source: 'auto',
-                                    attributes: {},
+                                    attributes: data.attributes.reduce(function(attrs_map: any, attr: any) {
+                                                                            attrs_map[attr.spec_id] = attr.value;
```

Comment:
as far as I understand, ``attr.spec_id`` is returned by nuclio function.
At the same time ``spec_id`` is the CVAT specific thing which additionally differs task by task. It means that nuclio function should not return these ids, only attribute names and matching should be implemented on CVAT side.

Reply:
Ok, it does make sense. And I guess the same is applicable for the changes in backend?

Reply:
Yep. Matching better to be implemented on backend (for both methods, when call automatic annotation for one frame and for a whole task)

Reply:
The only one question which I have so far is how to avoid attribute names and spec ids matching on client side in case of one frame auto annotation. 

What  I could figure out so far regarding attributes adding, is that attrs format expected to be like `{spec_id: value}` and checked [here](https://github.com/openvinotoolkit/cvat/blob/develop/cvat-core/src/annotations-collection.js#L722). So in `runInference` method I can't create some temporary attribute basing only on name and value, and then in the backend match label+attr_name coming from frontend into spec_id.

so my current implementation of [runInference](https://github.com/openvinotoolkit/cvat/blob/1c038bb4e983b120cbdd6ec6e2583146c1958f59/cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/tools-control.tsx#L1026) looks like that:
``` typescript
runInference={async (task: any, model: Model, body: object) => {
                    try {
                        this.setState({ mode: 'detection', fetching: true });
                        const result = await core.lambda.call(task, model, { ...body, frame })
                        let mapping = {};
                        task.labels.forEach((label: any) => {
                            mapping[label.name] = {};
                            label.attributes.forEach((attr: any) => {
                                mapping[label.name][attr.name] = attr.id;
                            })
                        });
                        let i = 0;
                        const states = result.map(
                            (data: any): any =>
                                new core.classes.ObjectState({
                                    shapeType: data.type,
                                    label: task.labels.filter((label: any): boolean => label.name === data.label)[0],
                                    points: data.points,
                                    objectType: ObjectType.SHAPE,
                                    frame,
                                    occluded: false,
                                    source: 'auto',
                                    attributes: data.attributes.reduce(function(attrs_map: any, attr: any) {
                                        attrs_map[mapping[data.label][attr.name]] = attr.value;
                                        return attrs_map;
                                    }, {}),
                                    zOrder: curZOrder,
                                }),
                        );
```
Finally, my questions 😄 
* Is it ok to keep mapping of attr_name returned from nuclio to spec_id in the FE?
* If it's ok to keep mapping in the FE, which approach is more preferable, to build `mapping` object from the code on render stage or on inference stage?

Reply:
@mikhail-treskin , could you please look how it is done for labels? For attributes we need to have mostly the same functionality. A serverless function should declare which attributes it supports (as it is done for labels). Based on the information, it is necessary map attributes per task.

Current patch cannot be applied as is to multiple tasks because spec_id will be different for different tasks.

Please let me know if you want to contribute the functionality. It can be quite complex if it is done in production quality. Also need to have a severless function which supports attributes. It will be an example and can be used for testing. I will move the patch into WIP for now.

Reply:
@nmanovic, yes the PR is definitely WIP6 just forgot to mark it properly.

Could you please have a look on changes in latest commit? as @bsekachev proposed previously i'm handling attributes returned from nuclio by name and then maps them into spec_id basing on db task data. 

> could you please look how it is done for labels? For attributes we need to have mostly the same functionality. A serverless function should declare which attributes it supports (as it is done for labels). Based on the information, it is necessary map attributes per task.

Does it presume that we also need to have functionality for attributes mapping in UI like it's done for labels?

> Also need to have a severless function which supports attributes

As an example can propose pipeline of few OMZ models like face_detection->emotion_recognition+age_gender_recognition (or only on of _recognition)

Reply:
@mikhail-treskin , the updated patch looks much better. We are more than happy to accept any contribution from our community. Also I will be happy to understand how you are using CVAT in your company and which features are missing. Probably it can be a meeting. Let me know if you are ready to share the information.

An example with OMZ models will be great. Could you please add one or several of them to the PR as serverless functions? If you have any requests to these models, please ping @snosov1. Probably he can help somehow. He also is responsible for OTE: https://github.com/openvinotoolkit/training_extensions

It will be great to map attributes in UI in the future. But let's consider the feature as an optional recommendation. It should not block the PR. I will merge it without the functionality.

Better to decelerate attributes in yml config for a serverless function. Thus it is easy to understand what the function supports.

If you can extend the tutorial: https://openvinotoolkit.github.io/cvat/docs/manual/advanced/serverless-tutorial/, it will be great. But it is an optional recommendation as well.

Reply:
@nmanovic could you please have a look on serverless function? 
I'm not sure regarding face-detection model, tried to choose between 0206 and 0205, looks like 0206 more accurate but much slower than 0205, not sure what is more preferable in case of automatic annotation scenario.